### ISSUE TITLE
Cannot SELECT on a hot standby coordinator

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#1496]
We try to obtain an AccessExclusiveLock during a SELECT, which errors out in recovery mode:

```
# select * from github_users limit 1;
ERROR:  cannot acquire lock mode AccessExclusiveLock on database objects while recovery is in progress
HINT:  Only RowExclusiveLock or less can be acquired on database objects during recovery.
```

This appears to be fixed on `master`, but not `6.2.3`.

(reported by Rakesh Therani on Slack)

**Comment by marcocitus** [comment]
This is fixed on `master`, but applies to `6.2.3`.

**Comment by marcocitus** [comment]
Fixed by f40f03270a10126f7742e481fe5ae18343c4c8e6
